httpcore5/src/main/java/org/apache/hc/core5/http/nio/entity/StringAsyncEntityProducer.java

```diff
@@ -54,6 +55,7 @@ public StringAsyncEntityProducer(
         super(bufferSize, fragmentSizeHint, contentType);
         Args.notNull(content, "Content");
         this.content = CharBuffer.wrap(content);
+        this.length = content.toString().getBytes(contentType.getCharset()).length;
```

Comment:
This is potentially very expensive in the case a large constant string is reused across many requests. Can we take some shortcuts, perhaps by validating the string is ascii, and using the input length in that case?

Reply:
I thought the implementation was missing, but it seems to be designed to produce HTTP requests without content-length header. Close this PR.